Bottom: 57122667224cf0ed01b47653c564e77c8ef59bc5
Top:    6b263bd401257b29cf6c38c88d7005b5f3a3fb98
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-06 00:22:51 +0100

Add a HID API call, notify_pcb_filename_change()

This is so the core can tell the GUI when the PCB being edited is saved
into a different file.


---

diff --git a/src/action.c b/src/action.c
index 9e47690..55825a9 100644
--- a/src/action.c
+++ b/src/action.c
@@ -5687,6 +5687,8 @@ ActionSaveTo (int argc, char **argv, Coord x, Coord y)
           SetChangedFlag (false);
           free (PCB->Filename);
           PCB->Filename = strdup (name);
+          if (gui->notify_filename_changed != NULL)
+            gui->notify_filename_changed ();
         }
       return 0;
     }
diff --git a/src/command.c b/src/command.c
index 009d373..963ed85 100644
--- a/src/command.c
+++ b/src/command.c
@@ -354,6 +354,8 @@ CommandSaveLayout (int argc, char **argv, Coord x, Coord y)
           SetChangedFlag (false);
           free (PCB->Filename);
           PCB->Filename = strdup (argv[0]);
+           if (gui->notify_filename_changed != NULL)
+            gui->notify_filename_changed ();
         }
       break;
 
diff --git a/src/hid.h b/src/hid.h
index d1c7c8e..2266b6c 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -577,6 +577,9 @@ typedef enum
      * our deliberate changes.
      */
     void (*notify_save_pcb) (const char *filename, bool done);
+
+    /* Notification to the GUI that the PCB file has been renamed. */
+    void (*notify_filename_changed) (void);
   };
 
 /* Call this as soon as possible from main().  No other HID calls are
